src/model/parseModelDefinition.ts

```diff
@@ -69,13 +70,25 @@ function deepParseModelDefinition<Dictionary extends ModelDictionary>(
       continue
     }
 
+    if (value instanceof NullableProperty) {
```

Comment:
Won't it have the same effect if we omit this `if` altogether? It should fall down to the very last `result.properties.push()`, doing the same thing. Let me know if I'm missing something here.

Reply:
The NullableProperty class gets caught by the `isObject` util since `typeof value === 'object'` where value is an instance of `NullableProperty` returns true. I could've written it as follows though:

```typescript
if (isObject<NestedModelDefinition>(value) && !(value instanceof NullableProperty)) {
  ...
}
```

The above is definitely clearer as to why it needs to be there but I kinda liked that `NullableProperty` was being handled explicitly which is why I went with what's there now. I'm very happy with either so up to you what you prefer 😄 .

There may be alternatives around changing the `isObject` util to be more specific but I don't know the implications of that.